### PR TITLE
ubuntu/jammy: fix lintian

### PR DESCRIPTION
**Pull request by aciba90**
do not squash

[SC-1154](https://warthogs.atlassian.net/browse/SC-1154)

### CODE REVIEW

**Review by blackboxsw — COMMENTED** [review]
Pending discussion on https://github.com/canonical/cloud-init/pull/1563 which is applicable here.

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Minor pending discussion on https://github.com/canonical/cloud-init/pull/1563 which applies here.

File: debian/cloud-init.lintian-overrides [hunk 1]
```diff
@@ -1,5 +1,7 @@
-cloud-init binary: systemd-service-file-refers-to-unusual-wantedby-target lib/systemd/system/cloud-config.service cloud-init.target
-cloud-init binary: systemd-service-file-refers-to-unusual-wantedby-target lib/systemd/system/cloud-final.service cloud-init.target
-cloud-init binary: systemd-service-file-refers-to-unusual-wantedby-target lib/systemd/system/cloud-init-local.service cloud-init.target
-cloud-init binary: systemd-service-file-refers-to-unusual-wantedby-target lib/systemd/system/cloud-init.service cloud-init.target
+# cloud-init.target should not be considered as unusual
+cloud-init binary: systemd-service-file-refers-to-unusual-wantedby-target cloud-init.target [lib/systemd/system/cloud-config.service]
+cloud-init binary: systemd-service-file-refers-to-unusual-wantedby-target cloud-init.target [lib/systemd/system/cloud-final.service]
+cloud-init binary: systemd-service-file-refers-to-unusual-wantedby-target cloud-init.target [lib/systemd/system/cloud-init-local.service]
+cloud-init binary: systemd-service-file-refers-to-unusual-wantedby-target cloud-init.target [lib/systemd/system/cloud-init.service]
```
Comment (paride) [inline debian/cloud-init.lintian-overrides:5]: Just fyi, if you want too many details on why this change is needed see:
- https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1007002
- https://lists.debian.org/debian-devel/2022/06/msg00184.html

This is a poorly documented change in the overrides format.

Comment (aciba90) [reply debian/cloud-init.lintian-overrides:5]: Very insightful links, thanks a lot!

**Comment by aciba90** [comment]
Pending: https://github.com/canonical/cloud-init/pull/1566#issuecomment-1177629649

**Comment by aciba90** [comment]
> Pending: [#1566 (comment)](https://github.com/canonical/cloud-init/pull/1566#issuecomment-1177629649)

Done, ready to be reviewed.

**Review by blackboxsw — APPROVED** [review]
separate commits look good
d/changelog format good
build-package/sbuild behavior good
We'll sort whether cloud-init package versioning departs from ~22.04.1 -> .22.04.1 separately from this PR.
